File: 2856-count-complete-subarrays-in-an-array/count-complete-subarrays-in-an-array.js
var countCompleteSubarrays = function(nums) {
    const totalDistinct = new Set(nums).size;
    let n = nums.length, count = 0;

    for (let i = 0; i < n; i++) {
        const seen = new Set();
        for (let j = i; j < n; j++) {
            seen.add(nums[j]);
            if (seen.size === totalDistinct) {
                count += n - j;
                break;
            }
        }
    }

    return count;
};

File: 2856-count-complete-subarrays-in-an-array/count-complete-subarrays-in-an-array.dart
class Solution {
  int countCompleteSubarrays(List<int> nums) {
    final totalDistinct = Set<int>.from(nums).length;
    int n = nums.length;
    int count = 0;
    
    for (int i = 0; i < n; i++) {
      Set<int> seen = {};
      for (int j = i; j < n; j++) {
        seen.add(nums[j]);
        if (seen.length == totalDistinct) {
          count += n - j;
          break;
        }
      }
    }
    
    return count;
  }
}